cmd/buildah/images.go

```diff
@@ -506,27 +539,37 @@ func formattedSize(size int64) string {
 	return fmt.Sprintf("%.3g %s", formattedSize, suffixes[count])
 }
 
-// reposToMap parses the specified repotags and returns a map with repositories
-// as keys and the corresponding arrays of tags as values.
-func imageReposToMap(repotags []string) map[string][]string {
-	// map format is repo -> tag
+// imageReposToMap parses the specified names list and returns a map with repositories
+// as keys and a slice of tags and digests as values.
+func imageReposToMap(names []string) map[string][]string {
+	// map format is repo -> []tagOrDigest
 	repos := make(map[string][]string)
-	for _, repo := range repotags {
-		var repository, tag string
-		if strings.Contains(repo, ":") {
-			li := strings.LastIndex(repo, ":")
-			repository = repo[0:li]
-			tag = repo[li+1:]
-		} else if len(repo) > 0 {
-			repository = repo
-			tag = "<none>"
-		} else {
+	for _, name := range names {
+		if name == "" {
 			logrus.Warnf("Found image with empty name")
+			continue
+		}
+		named, err := reference.ParseNormalizedNamed(name)
+		if err != nil {
+			logrus.Warnf("Error parsing name %q: %v", name, err)
+			continue
+		}
+		if named.Name() != name {
```

Comment:
This must be `.String()`.  `.Name()` only returns the repo.

Reply:
Ah, thanks!  Fixing.